src/hal/src/external_memory/mod.rs

```diff
@@ -0,0 +1,498 @@
+//! Structures related to the import external memory functionality
+
+mod errors;
+pub use errors::*;
+
+pub use external_memory::*;
+
+#[cfg(any(target_os = "linux", target_os = "android"))]
+pub use drm_fourcc::*;
+
+/// Buffer or image
+#[derive(Debug)]
+pub enum Resource<B: crate::Backend> {
+    /// Buffer
+    Buffer(B::Buffer),
+    /// Image
+    Image(B::Image),
+}
+
+/// External memory properties
+#[derive(Debug, PartialEq)]
+pub struct ExternalMemoryProperties {
```

Comment:
should this be in bitflags? or are we anticipating other non-bit properties in the future?

Reply:
Initially `ExternalMemoryProperties` got some other non boolean members, but due to the API shrink, they become useless and removed, but i hadn't noticed it can be turned into a bitflags. I will do that

Reply:
Done!